File: archive/logic/TuringMachineSimulationNew/Tapes/TapesUtilities/TapeConfig.cs
namespace TuringMachineSimulation.Logic.Tapes.TapesUtilities
{
    public class TapeConfig
    {
        public TapeTypes TapeType { get; set; }
        
        // Ignore if the tape type is infinite
        public int TapeLength { get; set; }
        
        // The string representation of the initial tape content
        public string TapeContent { get; set; }

        public TapeConfig(TapeTypes tapeType, int tapeLength, string tapeContent)
        {
            TapeType = tapeType;
            TapeLength = tapeLength;
            TapeContent = tapeContent;
        }
    }
}

File: frontend/turing-machine-web/src/logic/Tapes/TapesUtilities/TapeConfig.ts
import { TapeTypes } from "../TapeTypes"

export class TapeConfig
{
    public readonly TapeType: TapeTypes;
    public readonly TapeLength: number;
    public readonly TapeContent: string;

    /**
     * Constructor for configurating a tape.
     */
    constructor(tapeType: TapeTypes, tapeLength: number, tapeContent: string)
    {
        this.TapeType = tapeType;
        this.TapeLength = tapeLength;
        this.TapeContent = tapeContent;
    }

}